[CI] SpatialExtentAggregationNoLicenseIT testStExtentAggregationWithShapes failing

**Build Scans:**
- [elasticsearch-intake #20883 / part3](https://gradle-enterprise.elastic.co/s/i3h2tjpr6utro)
- [elasticsearch-periodic-platform-support #7287 / ubuntu-2404_platform-support-unix](https://gradle-enterprise.elastic.co/s/2dnlj7cy4zgey)
- [elasticsearch-periodic-platform-support #7287 / debian-12_platform-support-unix](https://gradle-enterprise.elastic.co/s/rrv2n7o6z6uf4)
- [elasticsearch-periodic-platform-support #7287 / oraclelinux-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/iekim5ysnn6h4)
- [elasticsearch-periodic #7265 / openjdk23_entitlements_true_checkpart3_java-matrix](https://gradle-enterprise.elastic.co/s/jrt2qrunwgjs2)
- [elasticsearch-pull-request #65024 / part-3](https://gradle-enterprise.elastic.co/s/l67z7jh7awoii)
- [elasticsearch-pull-request #64988 / part-3](https://gradle-enterprise.elastic.co/s/adipfrusuoouw)
- [elasticsearch-pull-request #64997 / part-3](https://gradle-enterprise.elastic.co/s/2y3dslnv7dofa)
- [elasticsearch-periodic-platform-support #7280 / rocky-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/qdzg4wt5d34ue)
- [elasticsearch-periodic #7258 / openjdk24_entitlements_false_checkpart3_java-matrix](https://gradle-enterprise.elastic.co/s/xxnm3mltr4cmo)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:internalClusterTest" --tests "org.elasticsearch.xpack.esql.spatial.SpatialExtentAggregationNoLicenseIT.testStExtentAggregationWithShapes" -Dtests.seed=8016E0FE4368F9AE -Dtests.locale=sd-Arab -Dtests.timezone=Pacific/Honolulu -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.spatial.SpatialExtentAggregationNoLicenseIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testStExtentAggregationWithShapes),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
junit.framework.AssertionFailedError: Expected exception VerificationException but no exception was thrown
```

**Issue Reasons:**
- [main] 2 consecutive failures in step rocky-9_platform-support-unix
- [main] 13 failures in test testStExtentAggregationWithShapes (8.4% fail rate in 154 executions)
- [main] 3 failures in step part-3 (4.8% fail rate in 63 executions)
- [main] 2 failures in step rocky-9_platform-support-unix (100.0% fail rate in 2 executions)
- [main] 3 failures in pipeline elasticsearch-pull-request (4.8% fail rate in 62 executions)
- [main] 3 failures in pipeline elasticsearch-periodic-platform-support (100.0% fail rate in 3 executions)
- [main] 2 failures in pipeline elasticsearch-periodic (66.7% fail rate in 3 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

OK. On further investigation, I can get this to fail on different branches with different seeds:

| seed | 8.x | 9.0 | main |
| === | === | === | === |
| -Dtests.seed=712BF32772B4874C |  FAIL |  FAIL | PASS |
| -Dtests.seed=17AC5C5D140EC9BF |  PASS |  PASS | FAIL |

My earlier tests with `-Dtests.iters=1000`, did pass with 0% failure on main, but re-testing later gave a high failure rate. And sometimes an early failure caused all further iterations to fail. A common theme not reported in the failure is the following assertion failure:

```
    java.lang.AssertionError: Request breaker not reset to 0 on node: node_s0
    Expected: <0L>
         but: was <240L>
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:964)
        at org.elasticsearch.test.InternalTestCluster.lambda$ensureEstimatedStats$35(InternalTestCluster.java:2472)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1498)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1471)
        at org.elasticsearch.test.InternalTestCluster.ensureEstimatedStats(InternalTestCluster.java:2470)
        at org.elasticsearch.test.TestCluster.assertAfterTest(TestCluster.java:181)
        at org.elasticsearch.test.InternalTestCluster.assertAfterTest(InternalTestCluster.java:2520)
        at org.elasticsearch.test.ESIntegTestCase.afterInternal(ESIntegTestCase.java:582)
        at org.elasticsearch.test.ESIntegTestCase.cleanUpCluster(ESIntegTestCase.java:2327)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
        at java.base/java.lang.reflect.Method.invoke(Method.java:565)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1763)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:1004)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
        at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
        at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
        at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
        at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
        at java.base/java.lang.Thread.run(Thread.java:1447)
```

--- a/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/spatial/SpatialExtentAggregationNoLicenseIT.java
+++ b/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/spatial/SpatialExtentAggregationNoLicenseIT.java
@@ -7,8 +7,11 @@
 
 package org.elasticsearch.xpack.esql.spatial;
 
+import org.elasticsearch.license.License;
+import org.elasticsearch.license.XPackLicenseState;
+import org.elasticsearch.license.internal.XPackLicenseStatus;
 import org.elasticsearch.plugins.Plugin;
-import org.elasticsearch.xpack.esql.action.EsqlPluginWithNonEnterpriseOrExpiredLicense;
+import org.elasticsearch.xpack.esql.plugin.EsqlPlugin;
 import org.elasticsearch.xpack.spatial.SpatialPlugin;
 
 import java.util.Collection;
@@ -18,11 +21,45 @@ public class SpatialExtentAggregationNoLicenseIT extends SpatialExtentAggregatio
 
     @Override
     protected Collection<Class<? extends Plugin>> nodePlugins() {
-        return List.of(SpatialPlugin.class, EsqlPluginWithNonEnterpriseOrExpiredLicense.class);
+        return List.of(TestSpatialPlugin.class, TestEsqlPlugin.class);
     }
 
     @Override
     public void testStExtentAggregationWithShapes() {
         assertStExtentFailsWith("index_geo_shape");
+    }
+
+    private static XPackLicenseState getLicenseState() {
+        License.OperationMode operationMode;
+        boolean active;
+        if (randomBoolean()) {
+            operationMode = randomFrom(
+                License.OperationMode.GOLD,
+                License.OperationMode.BASIC,
+                License.OperationMode.MISSING,
+                License.OperationMode.STANDARD
+            );
+            active = true;
+        } else {
+            operationMode = randomFrom(License.OperationMode.PLATINUM, License.OperationMode.ENTERPRISE, License.OperationMode.TRIAL);
+            active = false;  // expired
+        }
+
+        return new XPackLicenseState(
+            () -> System.currentTimeMillis(),
+            new XPackLicenseStatus(operationMode, active, "Test license expired")
+        );
+    }
+
+    public static class TestEsqlPlugin extends EsqlPlugin {
+        protected XPackLicenseState getLicenseState() {
+            return SpatialExtentAggregationNoLicenseIT.getLicenseState();
+        }
+    }
+
+    public static class TestSpatialPlugin extends SpatialPlugin {
+        protected XPackLicenseState getLicenseState() {
+            return SpatialExtentAggregationNoLicenseIT.getLicenseState();
+        }
     }
 }

